scripts/actions/fetch-observability-packs.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+/**
+ * This script is used to query the New Relic GraphQL API for Observability Packs.
+ * It then writes the array of Observability Packs to src/data/observability-packs.json
+ * It requires the following environment variables to be set:
+ * NR_GQL_URL - The New Relic GraphQL URL
+ * NR_API_TOKEN - A New Relic personal API token
+ **/
+
+/* eslint-disable no-console */
+const fs = require('fs');
+const fetch = require('node-fetch');
+
+const PACKS_FILE_PATH = './src/data/observability-packs.json';
+const NR_GQL_URL = process.env.NR_GQL_URL;
+const NR_API_TOKEN = process.env.NR_API_TOKEN;
+
+const packQuery = `# gql 
+  {
+    docs {
+      openInstallation {
+        observabilityPackSearch {
+          count
+          results {
+            observabilityPacks {
+              authors
+              dashboards {
+                description
+                name
+                screenshots
+                url
+              }
+              description
+              iconUrl
+              id
+              level
+              logoUrl
+              name
+              websiteUrl
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+/**
+ * Queries graphql for the provided query
+ * @param {String} queryString the graphql query to send
+ * @param {String} url NR graphql endpoint
+ * @param {String} token NR api token
+ * @returns {Promise<Object[]|undefined>} returns the resulting array
+ * or `undefined` if there was an error
+ **/
+const fetchPacks = async (queryString, url, token) => {
+  try {
+    const results = await fetch(url, {
+      method: 'post',
+      body: JSON.stringify({ query: queryString }),
+      headers: {
+        'Content-Type': 'application/json',
+        'Api-Key': token,
+      },
+    }).then((res) => res.json());
```

Comment:
We might want to consider avoiding `await` style syntax with `then` style syntax for promises.

An alternative would be to do something like:
```js
const res = await fetch();
const results = await res.json();
```

That would also allow us to more easily debug the response (e.g. log out the response code).

Reply:
Done!